packages/tsutils/src/types.ts

```diff
@@ -0,0 +1,7 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2022 Beyond Essential Systems Pty Ltd
+ */
+
+// TODO: Switch to 'Awaited' when upgrading to typescript 4.5+
+export type Resolved<T> = T extends Promise<infer R> ? R : T;
```

Comment:
This is a super common type, so wanted to move it here for re-use between packages.

In typescript 4.5 they introduced an `Awaited` type that does exactly this, so when we upgrade we can remove this and replace it